File: Lucy/LeetCode 75/Array_String/1431_Kids_With_the_Greatest_Number_of_Candies.js
/**
 * 1431. Kids With the Greatest Number of Candies
 *
 * url: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/?envType=study-plan-v2&envId=leetcode-75
 * difficulty: Easy
 * topic:
 * date: 2024.10.11(FRI)
 */

/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  const CANDIES_LEN = candies.length;
  const res = new Array(CANDIES_LEN);

  for (let i = 0; i < CANDIES_LEN; i++) {
    const calculated = candies[i] + extraCandies;
    const copied = [...candies];
    copied[i] = calculated;

    res[i] = calculated === Math.max(...copied);
  }

  return res;
};

console.log(kidsWithCandies([2, 3, 5, 1, 3], 3));

File: Seori/LeetCode 75/Array_String/1431. Kids With the Greatest Number of Candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        # candies 리스트의 원소들을 순회하면서, net_candies보다 크거나 같으면 문제 조건을 만족하므로 true, 그렇지 않으면 false를 반환한다.
        answer = []
        net_candies = max(candies) - extraCandies
        for candy in candies:
            if candy >= net_candies:
                answer.append(True)
            else:
                answer.append(False)
        
        return answer